src/tagmanager-wrap.js

```diff
@@ -0,0 +1,80 @@
+
+const defaultOptions = {
+  virtualPageViewEvent: 'virtual_pageview',
+  startPush: {
+    experiements: [],
+  },
+  addExperiments: [],
+};
+
+export default class TagManager {
+  constructor(dataLayer, params = defaultOptions) {
+    this.options = Object.assign(defaultOptions, params);
+    this.dataLayer = dataLayer || [];
+    this.addExperiments();
+    this.dataLayer.push(this.options.startPush);
+  }
+
+  addExperiments() {
+    Object.assign(this.options.startPush, {
+      experiments: this.experiments(),
+    });
+
+    return this;
+  }
+
+  experiments() {
+    return this.options.startPush.experiments.concat(this.options.addExperiments);
```

Comment:
@eduardojmatos If I'm not wrong, you could do something like this to preppend items to an array:

```
array = [newElement, ...oldArray]
```